src/controllers/redirect.controller.ts

```diff
@@ -43,15 +50,21 @@ export class RedirectController extends BaseController {
       return this.redirect(ENDORSE_COMPANY_CLOSURE_CERTIFICATE_URI)
     }
 
+    if (dissolution!.application_status === ApplicationStatus.PENDING_APPROVAL && this.getUserAlreadySigned(dissolution, userEmail)) {
+      return this.redirect(WAIT_FOR_OTHERS_TO_SIGN_URI)
+    }
+
     return this.redirect(ROOT_URI + '/')
   }
 
-  private getUserPendingSignature(dissolution: DissolutionGetResponse): Optional<DissolutionGetDirector> {
-    const userEmail: string = this.session.getUserEmail(this.httpContext.request)
-
+  private getUserPendingSignature(dissolution: DissolutionGetResponse, userEmail: string): Optional<DissolutionGetDirector> {
     return dissolution.directors.find(director => director.email === userEmail && !director.approved_at)
   }
 
+  private getUserAlreadySigned(dissolution: DissolutionGetResponse, userEmail: string): Optional<DissolutionGetDirector> {
+    return dissolution.directors.find(director => director.email === userEmail && director.approved_at)
+  }
```

Comment:
We are iterating over the same list multiple times. Is there a way we can either iterate over the list once, or could we even do this:
```
if (dissolution!.application_status === ApplicationStatus.PENDING_APPROVAL) {
  if (isUserPendingApproval(dissolution, userEmail)) {
    return this.redirect(ENDORSE_COMPANY_CLOSURE_CERTIFICATE_URI) 
  } else if (isApplicant) {
    return this.redirect(WAIT_FOR_OTHERS_TO_SIGN_URI)
  } else {
    // This is out of scope of your story btw - story is S4-38 for reference
    return this.redirect(CERTIFICATE_SIGNED_URI)
  }
}

...

private isUserPendingApproval(dissolution: DissolutionGetResponse, userEmail: string): boolean {
  return dissolution.directors.some(director => director.email === userEmail && !director.approved_at)
}
```